src/client/render/core.cpp

```diff
@@ -32,17 +32,24 @@ RenderingCore::RenderingCore(IrrlichtDevice *_device, Client *_client, Hud *_hud
 {
 	screensize = driver->getScreenSize();
 	virtual_size = screensize;
+
+	shadow_renderer = new ShadowRenderer(device, client);
 }
 
 RenderingCore::~RenderingCore()
 {
 	clearTextures();
+	if (shadow_renderer) {
+		delete shadow_renderer;
+		shadow_renderer = nullptr;
```

Comment:
you don't have to:
* test if the pointer is not nil, delete on nil is a NOOP
* nullify the pointer on the destructor, it's the  destructor :)

Reply:
The habits of an old programmer xD

Reply:
Done.